File: versioned_docs/version-1.18.x/examples/go/src/data_replication_mode.go
package main

import (
	"context"

	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	// Update replication mode without changing other settings
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Pause to queue transactions without sending them
	if err := client.SetReplicationMode(context.Background(), "my_replication", model.ReplicationModePaused); err != nil {
		panic(err)
	}

	// Disable to ignore new transactions
	if err := client.SetReplicationMode(context.Background(), "my_replication", model.ReplicationModeDisabled); err != nil {
		panic(err)
	}

	// Re-enable to resume sending queued data
	if err := client.SetReplicationMode(context.Background(), "my_replication", model.ReplicationModeEnabled); err != nil {
		panic(err)
	}
}

File: versioned_docs/version-1.20.x/examples/cpp/src/data_replication_mode.cc
#include <cassert>
#include <reduct/client.h>

using reduct::Error;
using reduct::IClient;
using reduct::ReplicationMode;

int main() {
    // Update replication mode without changing other settings
    auto client = IClient::Build("http://127.0.0.1:8383", {
        .api_token = "my-token"
    });

    // Pause to queue transactions without sending them
    auto err = client->SetReplicationMode("my_replication", ReplicationMode::kPaused);
    assert(err == Error::kOk);

    // Disable to ignore new transactions
    err = client->SetReplicationMode("my_replication", ReplicationMode::kDisabled);
    assert(err == Error::kOk);

    // Re-enable to resume sending queued data
    err = client->SetReplicationMode("my_replication", ReplicationMode::kEnabled);
    assert(err == Error::kOk);
}
